main.py

```diff
@@ -103,6 +103,11 @@ def batch_generator(testenc, seqlen, nsamples):
     emb_dev = emb.weight.device
     if emb_dev.type != "cuda":
         emb = emb.to(dev)
+        # opt has other embeddings
```

Comment:
I wold recommend to move this logic into `model_utils`; let's have a function there called `get_embeddings()`, returning `nn.ModuleDict` of embeddings and related things.

Reply:
Agree with the suggestion